toolkit/scripts/check_new_kernel_configs.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import json
+import argparse
+import sys
+import re
+from kernel_sources_analysis import get_data_from_config, extract_kernel_dir_name, extract_config_arch
+
+# Find the lines in the diff that contain +/- CONFIG
+# Ignore the lines that contain @
+# Return the set of configs that follow +/-
+def extract_modified_configs(input_string):
+    pattern = r'(?!.*@)[+-]\s*.*CONFIG.*'
```

Comment:
Try `^[-+](?:# )?(CONFIG_[^= ]+)`. It should grab the config names in the capture group.